cleanlab/datalab/internal/issue_manager_factory.py

```diff
@@ -79,23 +88,40 @@ class _IssueManagerFactory:
     """Factory class for constructing concrete issue managers."""
 
     @classmethod
-    def from_str(cls, issue_type: str) -> Type[IssueManager]:
+    def from_str(cls, issue_type: str, task: str = "classification") -> Type[IssueManager]:
         """Constructs a concrete issue manager class from a string."""
         if isinstance(issue_type, list):
             raise ValueError(
                 "issue_type must be a string, not a list. Try using from_list instead."
             )
-        if issue_type not in REGISTRY:
-            raise ValueError(f"Invalid issue type: {issue_type}")
+
+        # TODO: refactor it
+        if task is None:
+            if issue_type not in REGISTRY:
+                raise ValueError(f"Invalid issue type: {issue_type}")
+        else:
+            if task not in TASK_SPECIFIC_REGISTRY:
+                raise ValueError(
+                    f"Invalid task type: {task}, must be in {list(TASK_SPECIFIC_REGISTRY.keys())}"
+                )
+
+            if issue_type not in TASK_SPECIFIC_REGISTRY[task] and issue_type not in REGISTRY:
+                raise ValueError(f"Invalid issue type: {issue_type} for task {task}")
+
+            if issue_type in TASK_SPECIFIC_REGISTRY[task]:
+                REGISTRY[issue_type] = TASK_SPECIFIC_REGISTRY[task][issue_type]
```

Comment:
```suggestion
        if task is None and issue_type not in REGISTRY:
            raise ValueError(f"Invalid issue type: {issue_type}")
        if task not in TASK_SPECIFIC_REGISTRY:
            raise ValueError(
                f"Invalid task type: {task}, must be in {list(TASK_SPECIFIC_REGISTRY.keys())}"
            )
        if issue_type not in TASK_SPECIFIC_REGISTRY[task] and issue_type not in REGISTRY:
            raise ValueError(f"Invalid issue type: {issue_type} for task {task}")
```


You can remove most of the nesting this way. If you feel these are too many if-statements, consider extracting all of them into an internal method, like

```python
@classmethod
def _validate_arguments(cls, issue_type: str, task: str) -> None:
    ...
```

Also, we shouldn't mutate the `REGISTRY` with `from_str`. We'd just return the `TASK_SPECIFIC_REGISTRY[task][issue_type]` instead.

Reply:
Totally agree, refactoring was on todo list :)

Reply:
Just a reminder to handle the case where we return the task-specific issue manager:

```python
TASK_SPECIFIC_REGISTRY[task][issue_type]
```